src/dolphin/workflows/displacement.py

```diff
@@ -315,3 +342,53 @@ def _print_summary(cfg):
     logger.info(f"Maximum memory usage: {max_mem:.2f} GB")
     logger.info(f"Config file dolphin version: {cfg._dolphin_version}")
     logger.info(f"Current running dolphin version: {__version__}")
+
+
+def run_timeseries(
+    ts_opts: TimeseriesOptions,
+    unwrapped_paths: Sequence[Path],
+    conncomp_paths: Sequence[Path],
+    cor_paths: Sequence[Path],
+    stitched_amp_dispersion_file: Path,
+    num_threads: int = 5,
+) -> list[Path]:
+    """Invert the unwrapped interferograms, estimate timeseries and phase velocity."""
+    output_path = ts_opts._directory
+    output_path.mkdir(exist_ok=True, parents=True)
+
+    ifg_date_pairs = [get_dates(f) for f in unwrapped_paths]
+    sar_dates = sorted(set(flatten(ifg_date_pairs)))
+    # if we did single-reference interferograms, for `n` sar dates, we will only have
+    # `n-1` interferograms. Any more than n-1 ifgs means we need to invert
+    needs_inversion = len(unwrapped_paths) > len(sar_dates) - 1
+    # check if we even need to invert, or if it was single reference
+    if needs_inversion:
+        logger.info("Selecting a reference point for unwrapped interferograms")
+        reference = timeseries.select_reference_point(
+            conncomp_paths,
+            stitched_amp_dispersion_file,
+            output_dir=ts_opts._directory,
+        )
```

Comment:
Shouldn't this be outside? regardless if we invert a network or not, the time-series should be referenced before velocity estimation.

Reply:
I can do that, I was picturing that you only definitely need to reference it for the inversion, but the velocity map can be adjusted later

Reply:
no wait what am I talking about, you definitely need to reference before velocity :face_with_spiral_eyes:  thanks for catching